### PR TITLE
Schedule decision task using big graph scheduler

### PR DESCRIPTION
**Pull request by gregarndt**

### CODE REVIEW

**Comment by gregarndt** [comment]
@djmitche I think this is what we were looking for with scheduling a decision task using the queue rather than scheduler.  I think that if this is ok and lands, then we could start passing this info to the command line for the decision task so that it can create tasks with the same parameters, such as task group ID.

File: src/jobs/taskcluster_graph.js [hunk 1]
```diff
@@ -92,63 +107,133 @@ export default class TaskclusterGraphJob extends Base {
...
       console.log(`Fetching url ${graphUrl} for '${repo.alias}' push id ${push.id}`);
-      graphText = await fetchGraph(job, graphUrl);
+      let graphText = await fetchGraph(job, graphUrl);
+      templateVariables.source = graphUrl;
+
+      let scheduler = new taskcluster.Scheduler({
+        credentials: this.config.taskcluster.credentials,
+        // Include scopes for creating and extending task graphs.
+        authorizedScopes: scopes.concat(['scheduler:create-task-graph', 'scheduler:extend-task-graph:*'])
+      });
```
Comment (djmitche) [inline src/jobs/taskcluster_graph.js]: Do we still need `scheduler`?  It shouldn't be involved in a big-graph-scheduler operation..

Comment (djmitche) [inline src/jobs/taskcluster_graph.js]: oh, I see, this is the "old" branch.  I suppose it can't hurt to keep it around..

**Comment by djmitche** [comment]
I like this.  I don't know if the "backward compatible" bits will work, but it can't hurt to have them here.

Landing this will mean that decision tasks are immediately scheduled using the big-graph-scheduler.  I think that will still work (although the decision tasks will create a second task graph).  And I think the existing retriggering support will continue to work.  So we should be OK!

File: src/jobs/taskcluster_graph.js (hunk 1 above)
Comment (gregarndt) [inline src/jobs/taskcluster_graph.js]: I left it like this for branches like esr45 that produce a graph.json and the worker extends the graph.  If a graph was never created with the original decision task, there is no graph to extend.

**Comment by gregarndt** [comment]
Yea I I think retriggering is ok, and this is the first step into the changes.  The next step would be for the decision task to create the tasks within the same group so that everything lives nice and cosy.

**Comment by gregarndt** [comment]
I have been waiting for the trees to reopen before landing this so I didn't cause more problems if something went wrong but they're still closed.  I am signing off for the night, will be traveling back home tomorrow, and then I'm on PTO.  if you want to land this on Monday, I can make myself available to hopefully see things go ok.

**Comment by gregarndt** [comment]
Also, as far as the group ID and decision task ID being the same, is it safe to assume that the decision task is the only task within .taskcluster.yml?  If so, then instead of looping over tasks in .taskcluster.yml we will assume there is just one task and schedule tasks[0] as the decision task.

**Comment by djmitche** [comment]
I think that's a safe assumption, yes.  I'll get it landed and, if things explode, back it out -- no need to cancel your PTO :)

**Comment by jonasfj** [comment]
>  If so, then instead of looping over tasks in .taskcluster.yml we will assume there is just one task and schedule tasks[0] as the decision task.

@djmitche if the `.taskcluster.yml` can contain more than one task... I suggest we change the format so it only contains one task.

**Comment by djmitche** [comment]
I'd prefer to keep the format compatible with tc-github, which allows multiple tasks.  I think we can say that the taskGroupId will match the taskId of the first task in taskcluster.yml without loss of too much generality (this is slightly different than what greg suggested)

**Comment by djmitche** [comment]
Closed in favor of #87

**Comment by jonasfj** [comment]
Yes, taskId = taskGroupId for the first task... that seems solid! :)

--
Regards Jonas Finnemann Jensen.

2016-07-20 10:02 GMT-07:00 Dustin J. Mitchell <notifications@github.com>:

**Pull request closed without merging**